### ISSUE TITLE
Ingame debugging

### ISSUE THREAD

**Issue by krille90** [unitystation/unitystation#1574]
# Description
Currently, game logic is hardcoded and can't be controlled ingame. This is kind of annoying when developing and debugging. 

A solution would be to add an ingame debug menu, maybe attached to the F5 menu.

## Required Features
- [x] Player respawning
- [x] Round restarting
- [x] Teleporting (to position, player, matrix, ...)
- [x] Role changing
- [x] Shuttle calling
- [x] Suicide
- [ ] God mode
- [x] Spawning items
- [x] Spawning dummy player (and eventually NPC)
- [x] Taking control of different bodies 
- [ ] Spawning / removing atmos gas
- [ ] Door permission changing
- [x] Apply damage to different body parts

## Additional Features
- Only possible as Server player / in Debug mode (or admin, which don't exist yet, I think)
- Code should be easily extendable to add more features

### Remarks
Feel free to add more in the comments.

**Comment by Jibbajabbafic** [comment]
Some other features that might be useful: heal/damage body parts, toggle damage taking (can be specific damage types, or all of them), object spawning (needs #1488), dummy player (and eventually NPC) spawning, taking control of different bodies and atmos gas spawning

**Comment by Jibbajabbafic** [comment]
If this is available to admins we'll also need kicking, banning (permanent and temporary), unbanning, permission changing and admin PM *bwoink*

**Comment by Jibbajabbafic** [comment]
Although it would be worth separating admin and debug menus at some point

**Comment by krille90** [comment]
Yea, let's focus first on debug features

**Comment by donovanhubbard** [comment]
I actually started working on this yesterday. I put a suicide command in the in game debug menu. I'll get that polished off and committed.

**Comment by DooblyNoobly** [comment]
yep idea is to have them as admin commands that are typed into the console input. So good idea @fuele , make it so it's easy to add more commands in the future and make sure to check that its the Server / Host before allowing the command to be executed.

**Comment by donovanhubbard** [comment]
The groundwork for this has been laid in the in game debugger. Documentation for adding new commands is here. https://github.com/unitystation/unitystation/wiki/How-to-use-the-in-game-debugger

**Comment by daalta** [comment]
Xeno NPCs can be spawned in the current version, so this issue could be updated.

**Comment by MaxIsJoe** [comment]
All this is technically doable nowadays.
Some are less straight-forward than others due to the lack of dedicated tools for specific things listed here, but you *can* do everything here using VV or different admin tools.
